zproject/urls.py

```diff
@@ -565,37 +565,11 @@
     url(r'^terms/$', zerver.views.portico.terms_view, name='terms'),
     url(r'^privacy/$', zerver.views.portico.privacy_view, name='privacy'),
 
-    url(r'^config-error/google$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'backend_error': True, 'social_backend_name': 'google'},),
-    url(r'^config-error/github$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'backend_error': True, 'social_backend_name': 'github'},),
-    url(r'^config-error/gitlab$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'backend_error': True, 'social_backend_name': 'gitlab'},),
-    url(r'^config-error/smtp$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'smtp_error': True},),
-    url(r'^config-error/ldap$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'ldap_error_realm_is_none': True},
-        name='ldap_error_realm_is_none'),
-    url(r'^config-error/dev$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'dev_not_supported_error': True},
-        name='dev_not_supported'),
-    url(r'^config-error/saml$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'saml_error': True},),
-    url(r'^config-error/remoteuser/backend_disabled$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'remoteuser_error_backend_disabled': True},),
-    url(r'^config-error/remoteuser/remote_user_header_missing$', TemplateView.as_view(
-        template_name='zerver/config_error.html',),
-        {'remoteuser_error_remote_user_header_missing': True},),
```

Comment:
Wouldn't it be viable and simpler to instead have ``url(r'^config-error/([\w,-]+), view_function_handling_this, ...`` instead of adding these urls one-by-one in a for loop?
Similarly to how we have
```
    url(r'^accounts/login/social/([\w,-]+)$', zerver.views.auth.start_social_login,
        name='login-social'),
    url(r'^accounts/login/social/([\w,-]+)/([\w,-]+)$', zerver.views.auth.start_social_login,
        name='login-social-extra-arg'),
```

for social login with all these backends?

Reply:
Maybe we should start writing new urls using the django2 ``path`` - https://docs.djangoproject.com/en/2.2/ref/urls/#path :thinking: We should convert the old ``url`` entries to use ``path`` too at some point, but that's going to be a hassle :confounded:

Reply:
> Wouldn't it be viable and simpler to instead have `url(r'^config-error/([\w,-]+$', view_function_handling_this, ...` instead of adding these urls one-by-one in a for loop?

It would be great if we could do that. But the problem is with handling context variables, also we cannot give different names to each error page, which is a major issue because we use `reverse` at some places. We can reduce one variable `social_backend_name` by changing it to `url(r'^config-error/(?P<social_backend_name>[\w,-]+)$', ......`.  but it becomes difficult to handle `has_markdown_file` variable I added in this commit.